var/spack/repos/builtin/packages/py-alphafold/package.py

```diff
@@ -17,31 +17,38 @@ class PyAlphafold(PythonPackage, CudaPackage):
     url = "https://github.com/deepmind/alphafold/archive/refs/tags/v2.1.1.tar.gz"
     maintainers = ["aweits"]
 
+    version("2.2.4", sha256="8d756e16f6dc7897331d834aade8493820d0ff6a03bf60ce511bac4756c1b1e8")
     version("2.1.1", sha256="1adb6e213ba9ac321fc1acb1c563ba9b4fc054c1cebe1191bc0e2aaa671dadf7")
 
     conflicts("platform=darwin", msg="alphafold is only supported on Linux")
 
     depends_on("python@3.7:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("py-absl-py@0.13.0:", type=("build", "run"))
+    depends_on("py-absl-py@0.13.0:", type=("build", "run"), when="@2.1.1")
+    depends_on("py-absl-py@1.0.0:", type=("build", "run"), when="@2.2.4")
     depends_on("py-biopython@1.79:", type=("build", "run"))
     depends_on("py-chex@0.0.7:", type=("build", "run"))
-    depends_on("py-dm-haiku@0.0.4:", type=("build", "run"))
+    depends_on("py-dm-haiku@0.0.4:", type=("build", "run"), when="@2.1.1")
+    depends_on("py-dm-haiku@0.0.7:", type=("build", "run"), when="@2.2.4")
     depends_on("py-dm-tree@0.1.6:", type=("build", "run"))
```

Comment:
I'd add it if we were actually building the docker image - I'm stopping at just spackify-ing the build

Reply:
`setup.py` contains `docker`, so if I were to `pip install alphafold`, it would install docker. I don't think it's a good idea to diverge too much from the listed required deps. Things like `pip check` will fail if deps declared as required are missing.